File: nomad-samples/js/asset-groups/asset-groups/add-asset-to-asset-group.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function addAssetToAssetGroup(authToken) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);
  
    const BODY = [
        "9e92b187-745d-4c35-8265-088415dea57b"
    ];
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/assetgroup/add/{assetGroupId}`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Failed to add asset to asset group");
}

File: nomad-samples/python/asset_groups/asset_group/add_asset_to_asset_group.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

def add_asset_to_asset_group(AUTH_TOKEN: str, ASSET_GROUP_ID) -> dict:
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = PORTAL_API_URL + "/assetgroup/add/" + ASSET_GROUP_ID

    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }
    
    BODY = [
        "9e92b187-745d-4c35-8265-088415dea57b"
    ]

    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)
   	
    except:
      	api_exception_handler(RESPONSE, "Add asset to asset group failed")